### PR TITLE
Use mixer client opaque config to configure source and destination

### PR DESCRIPTION
**Pull request by kyessenov**
Configures mixer client filter to send `source.uid` and `target.uid` to be source and destination pod IPs. This is not working for source-to-source traffic when source and target are the same.

Note: you need to change discovery service to take mixer address as a parameter in demos.

### CODE REVIEW

File: proxy/envoy/testdata/cds-circuit-breaker.json.golden [hunk 1]
```diff
@@ -71,4 +71,4 @@
     }
    }
   ]
- }
+ }
```
Comment (andraxylia) [inline proxy/envoy/testdata/cds-circuit-breaker.json.golden]: Add newline

File: proxy/envoy/testdata/cds.json.golden [hunk 2]
```diff
@@ -29,4 +29,4 @@
     "lb_type": "round_robin"
    }
   ]
- }
+ }
```
Comment (andraxylia) [inline proxy/envoy/testdata/cds.json.golden]: Add newline

File: proxy/envoy/policy.go [hunk 3]
```diff
@@ -18,20 +18,39 @@
...
-		for _, f := range l.Filters {
-			if f.Name == HTTPConnectionManager {
-				http := (f.Config).(*HTTPFilterConfig)
-				http.Filters = append([]HTTPFilter{{
-					Type:   "both",
-					Name:   "mixer",
-					Config: &FilterMixerConfig{MixerServer: mixer},
-				}}, http.Filters...)
+func insertMixerFilter(listeners []*Listener, context *ProxyContext) {
+	if context.MeshConfig.MixerAddress != "" {
```
Comment (andraxylia) [inline proxy/envoy/policy.go]: You can return if MixerAddress == "", and have nicer code style.

File: test/integration/manager-discovery.yaml.tmpl [hunk 4]
```diff
@@ -27,7 +27,7 @@ spec:
       - name: manager
         image: {{.hub}}/runtime:{{.tag}}
         imagePullPolicy: Always
-        args: ["discovery", "-v", "{{.verbosity}}", "-n", "{{.namespace}}"]
+        args: ["discovery", "-v", "{{.verbosity}}", "-n", "{{.namespace}}", "-m", "mixer:9091"]
```
Comment (andraxylia) [inline test/integration/manager-discovery.yaml.tmpl]: Why is this needed in the manager?

Comment (kyessenov) [inline test/integration/manager-discovery.yaml.tmpl]: RDS needs to know about existence of mixer.

File: proxy/envoy/testdata/cds.json.golden (hunk 2 above)
Comment (kyessenov) [inline proxy/envoy/testdata/cds.json.golden]: autogenerated file?
